packages/core/src/actions/execute.ts

```diff
@@ -73,73 +153,31 @@ export const executeTask = async (args: {
       isSetImplementationAction(fromRawChugSplashAction(action.action))
     )
 
-    // We execute the SetStorage actions in batches, which have a size based on the maximum
-    // block gas limit. This speeds up execution considerably. To get the batch size, we divide
-    // the block gas limit by 150,000, which is the approximate gas cost of a single SetStorage
-    // action. We then divide this quantity by 2 to ensure that we're not approaching the block
-    // gas limit. For example, a network with a 30 million block gas limit would result in a
-    // batch size of (30 million / 150,000) / 2 = 100 SetStorage actions.
-    const { gasLimit: blockGasLimit } = await executor.provider.getBlock(
-      'latest'
-    )
-    const batchSize = blockGasLimit.div(150_000).div(2).toNumber()
-
     // Execute SetStorage actions in batches.
-    const setStorageActions = bundle.actions.slice(0, firstDeployImplIndex)
-    for (
-      let i = currActionsExecuted;
-      i < firstDeployImplIndex;
-      i += batchSize
-    ) {
-      const setStorageBatch = setStorageActions.slice(i, i + batchSize)
-      await (
-        await chugSplashManager.executeMultipleActions(
-          setStorageBatch.map((action) => action.action),
-          setStorageBatch.map((action) => action.proof.actionIndex),
-          setStorageBatch.map((action) => action.proof.siblings)
-        )
-      ).wait()
-      currActionsExecuted += setStorageBatch.length
-    }
-
-    logger.info(
-      `State variables have been set. Deploying the implementation contracts...`
+    logger.info(`executing SetStorage actions...`)
+    await executeBatchActions(bundle.actions.slice(0, firstDeployImplIndex))
+    logger.info(`executed SetStorage actions`)
+
+    // Execute DeployImplementation actions in batches.
+    logger.info(`executing DeployImplementation actions...`)
+    await executeBatchActions(
+      bundle.actions.slice(firstDeployImplIndex, firstSetImplIndex)
     )
-
-    // Execute DeployImplementation actions in series. We execute them one by one since each one
-    // costs significantly more gas than a setStorage action (usually in the millions).
-    for (let i = currActionsExecuted; i < firstSetImplIndex; i++) {
-      const action = bundle.actions[i]
-      await (
-        await chugSplashManager.executeChugSplashAction(
-          action.action,
-          action.proof.actionIndex,
-          action.proof.siblings
-        )
-      ).wait()
-      currActionsExecuted += 1
-      logger.info(
-        `Deployed implementation contract: ${
-          currActionsExecuted - firstDeployImplIndex
-        }/${firstSetImplIndex - firstDeployImplIndex}`
+    logger.info(`executed DeployImplementation actions`)
+
+    // Execute SetImplementation actions in batches.
```

Comment:
nit: We execute the SetImplementation actions in a single transaction (not in batches)